hera_cal/io.py

```diff
@@ -713,7 +716,10 @@ def iterate_over_bls(self, Nbls=1, bls=None, chunk_by_redundant_group=False, red
             # make sure that every baseline is in reds
             baseline_chunks = chunk_baselines_by_redundant_groups(reds=reds, max_chunk_size=Nbls)
         for chunk in baseline_chunks:
-            yield self.read(bls=chunk)
+            if freqs_to_load is None:
```

Comment:
Do you actually need this if / else? If freqs is None, won't self.read behave appropriately?

Reply:
you're right, if freqs i None, things will work fine.